File: 0205/205.isomorphic-strings.java
import java.util.HashMap;

import javax.imageio.plugins.tiff.TIFFField;

/*
 * @lc app=leetcode id=205 lang=java
 *
 * [205] Isomorphic Strings
 */

// @lc code=start
class Solution {
    public boolean isIsomorphic(String s, String t) {

        String[] s1 = s.split("");
        String[] t1 = t.split("");

        HashMap<String, String> results = new HashMap<>();

        for (int k = 0; k < s1.length; k++) {
            if (!results.containsKey(s1[k])) {
                if (!results.containsValue(t1[k])) {
                    results.put(s1[k], t1[k]);
                } else {
                    return false;
                }
            } else {
                if (!results.get(s1[k]).equals(t1[k])) {
                    return false;
                }
            }
        }

        return true;

    }

    // public static void main(String[] args) {
    //     var sol = new Solution();
    //     String s = "egg";
    //     String t = "add";
    //     sol.isIsomorphic(s, t);
    // }
}
// @lc code=end

File: 0205/205.isomorphic-strings.py
#
# @lc app=leetcode id=205 lang=python3
#
# [205] Isomorphic Strings
#

# @lc code=start
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        s1 = list([*s])
        t1 = list([*t])

        results = {}

        for index, value in enumerate(s1):
            if s1[index] not in results:
                if t1[index] not in results.values():
                    results.update({s1[index]: t1[index]})
                else:
                    return False
            elif results.get(s1[index]) != t1[index]:
                    return False

        return True


# @lc code=end
